### DISCUSSION TITLE
PID 大作业提交文件格式

### DISCUSSION THREAD

**Discussion by AuroraRyan** [physics-data/faq#172] [Q&A] [answered]
问题1：上传的文件大小是否有限制？需要提交多少个event?int16只有3万多个事件，problem.h5有十万多个event，in32直接1.7M上传不上去

问题2：h5文件表格是否需要标签（列名）？平台提交文件无法评测
提交文件显示：`Error grading your submission: argument of type 'NoneType' is not iterable`

我使用 `h5py.File("file_name", 'w')` 的方法保存numpy数组，不知道如何修改列名，使用pandas，转换成dataframe修改列名无法保存为单独一个表格（在一个group下），而且保存后会有列名错误
不知道助教使用了怎样的函数

**Comment by Harry-Chen** [comment, marked as answer]
我回答一部分平台相关的问题：

1. 大小限制非常大，1.7MB 的文件不会由于大小限制无法上传。
2. 请严格按照数据格式，我们给出了一个 `example.h5`，任何不按照格式生成的答案都无法被评分。

**Reply by wengjun232** [reply]
答案的大小应该遵循平台的限制。

提交的内容包括事例的ID和alpha粒子的概率。一共是>10万个事例。具体的格式可以看`example.h5`。

**Comment by heroxbd** [comment]
可以使用 `numpy` 的 `structured array` 保存有列名的 `dataset`。`pandas` 中有 `to_records` 函数来把 `DataFrame` 转化成 `structured array`。
